x-pack/legacy/plugins/reporting/test_helpers/create_mock_server.ts

```diff
@@ -4,9 +4,32 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ServerFacade } from '../server/types';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { createHttpServer, createCoreContext } from 'src/core/server/http/test_utils';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { coreMock } from 'src/core/server/mocks';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { ContextService } from 'src/core/server/context/context_service';
 
-export const createMockServer = (): ServerFacade => {
-  const mockServer = {};
-  return mockServer as any;
+const coreId = Symbol('reporting');
+
+export const createMockServer = async () => {
+  const coreContext = createCoreContext({ coreId });
+  const contextService = new ContextService(coreContext);
+
```

Comment:
Is this a direct (untouched) copy of `setupServer` in `src/core/server/saved_objects/routes/integration_tests/test_utils.ts` ?

If so, you might want to directly import it from there instead of duplicating it. We should expose this as a public test API in a near futur anyway.